internal/safefile/safeopen.go

```diff
@@ -340,6 +340,35 @@ func MkdirRelative(path string, root *os.File) error {
 	return err
 }
 
+// MkdirAllRelative creates each directory in the path relative to a root, failing if
+// any existing intermediate path components are reparse points.
+func MkdirAllRelative(path string, root *os.File) error {
+	pathParts := strings.Split(filepath.Clean(path), (string)(filepath.Separator))
+	for index := range pathParts {
+
+		partialPath := filepath.Join(pathParts[0 : index+1]...)
+		stat, err := LstatRelative(partialPath, root)
+
+		if err != nil && os.IsNotExist(err) {
+			if err := MkdirRelative(partialPath, root); err != nil {
+				return err
+			}
+			continue
+		}
+
+		if err != nil {
+			return err
+		}
```

Comment:
```go
       if err != nil {
		if os.IsNotExist(err) {
			if err := MkdirRelative(partialPath, root); err != nil {
				return err
			}
			continue
		}
		return err
	}
```

This seems easier to read (at least to me)

Reply:
Makes sense. I'll try and make that change tonight. I think I originally wanted
```Go
if err != nil && os.IsNotExist(err) {
// Mkdir and recovery
} else if err != nil {
  return err
}
```
but Go syntax or the linter didn't like that.